package.json

```diff
@@ -1,9 +1,9 @@
 {
   "name": "protocol-common",
-  "version": "1.0.3",
+  "version": "1.0.4",
```

Comment:
I know its not ready for review yet.    Adding now to keep it fresh in the memory.

The version number should more than a minor update here. due to the significance of these changes   Maybe a 2.x.x but at least 1.1.x

Reply:
Yeah, this is a really good point and one I struggled with. From a high level point of view, this is about making the major version increment from 0.x.x to 1.x.x actually usable. As in, that change already introduced the need to use ESM, while these changes are what make it possible to use ESM. So the way I though of it was that this is effectively just a bug fix within the 1.x.x set of changes.

That said, I did go a bit farther and introduce some other changes that make the ESM version nice to use, so I could see a 1.1.x update making sense. The reason I didn't do that is because this is a somewhat experimental update given that I haven't been able to test it against the whole system yet. I was thinking that once it's been tested and finalized and the documentation has been updated, that might justify the 1.1.x upgrade.

Reply:
lets go with 1.1.x then.